trin-core/src/portalnet/types/content_key.rs

```diff
@@ -1,14 +1,39 @@
+use crate::portalnet::types::messages::ByteList;
 use anyhow::anyhow;
 use ethereum_types::{U256, U512};
 use sha2::{Digest as Sha2Digest, Sha256};
 use sha3::{Digest, Keccak256};
 use ssz::{self, Decode, Encode};
 use ssz_derive::{Decode, Encode};
 use ssz_types::{typenum, FixedVector, VariableList};
+use std::ops::Deref;
 
 /// SSZ encoded overlay content key as bytes
 pub type RawContentKey = Vec<u8>;
 
+/// uTP content payload list represented as SSZ encoded Container[payloads: List[ByteList, max_length=64]]
+#[derive(Clone, Debug, Encode, Decode)]
+pub struct ContentPayloadList {
+    payloads: Vec<ByteList>,
+}
```

Comment:
We can replace with [`VariableList`](https://docs.rs/eth2_ssz_types/0.2.1/ssz_types/struct.VariableList.html).